client/src/app/components/ModalProps.tsx

```diff
@@ -15,5 +15,10 @@ export interface ModalProps {
     data: {
```

Comment:
I this file, I have my doubts about `ModalIcon`, I think is too restrictive and it forces you to update this enum each time you want to add a new possible icon to the modal toggle. I would remove this and allow passing any material icon in a string format. If tomorrow you want to have a modal that is triggered by a smiley face, this code wouldn't allow it